fsw/cfe-core/src/inc/ccsds.h

```diff
@@ -122,6 +122,12 @@ typedef struct {
 
    uint8  Time[CCSDS_TIME_SIZE];
 
+#if ( CCSDS_TIME_SIZE == 6 )
+   uint8  tempPad1[4];
+#elif ( CCSDS_TIME_SIZE == 8 )
+   uint8  tempPad1[2];
+#endif
```

Comment:
Padding shouldn't be added this way -- it isn't a fixed quantity.  Furthermore the "ccsds.h" definition should be the source of truth -- that is, it should be the actual definition of the structure without getting into any architecture-specific padding requirements.

The other reason not to do it this way is that the opposite problem exists when extended headers is enabled.  This adds 4 bytes to the primary, so now the size of TLM hdr becomes 16 and the CMD header becomes 12, so now CMDs will need padding and TLM packets will not.

The paradigm should be that `ccsds.h` definitions represent the basic fundamental primitives, and `cfe_sb.h` structures represent how these are actually realized for use within the software bus and applications.

So - in the related PR #678 the `CFE_SB_TlmHdr_t`  becomes a padded and aligned version of `CCSDS_TlmSecHdr_t`

One can reliably determine the actual amount of padding on the current system by checking `sizeof(CFE_SB_TlmHdr_t) - sizeof(CCSDS_TlmSecHdr_t)` or equivalent...

This will work on 32 bit or 64 bit, with or without extended headers, and should continue to work even if the headers are changed in the future.

Reply:
or what about using a (surrounding) union, like:

 `typedef union { uint8 Time[CCSDS_TIME_SIZE], TotalSz[10] } CCSDS_TlmSecHdr_t;`